compiler/rustc_mir/src/interpret/traits.rs

```diff
@@ -121,4 +121,27 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         }
         Ok((Size::from_bytes(size), align))
     }
+
+    pub fn read_new_vtable_after_trait_upcasting_from_vtable(
+        &self,
+        vtable: Pointer<Option<M::PointerTag>>,
+        idx: u64,
+    ) -> InterpResult<'tcx, Pointer<M::PointerTag>> {
+        let pointer_size = self.pointer_size();
+
+        let vtable = self
+            .memory
+            .get(
+                vtable,
+                pointer_size * idx.checked_add(1).unwrap(),
+                self.tcx.data_layout.pointer_align.abi,
+            )?
+            .expect("cannot be a ZST");
```

Comment:
For consistency, can you write this code like `get_vtable_slot`?
Currently you are conceptually reading the first `idx` entries of the vtable, but its more accurate IMO to just read the one you actually need.

So, something like
```rust
let vtable_slot = vtable.offset(ptr_size * idx, self)?;
let vtable_slot = self
            .memory
            .get(vtable_slot, ptr_size, self.tcx.data_layout.pointer_align.abi)?
.expect("cannot be a ZST");
```

Reply:
Sure. Done.